.github/workflows/pr_build.yml

```diff
@@ -1,83 +1,93 @@
+# PR-build-CI.yaml
+# This yaml script will trigger when a PR is created on to the main branch - will ensure that the repo is built successfully with all the dependencies.
+## Platform - Ubuntu
 name: Azure WindUp Rulesets PR builder
 
+env:
+    JAVA_DISTRIBUTION: 'microsoft'
+
 on:
-  workflow_dispatch:
-  pull_request:
-    branches:
-      - master
-      - 'releases/**'
+    workflow_dispatch:
+    pull_request:
+        branches:
+            - main
 
 jobs:
+    windup-build:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout JEG windup Azdo repo
+              run: 'git clone https://${{secrets.USER}}:${{secrets.GH}}@devdiv.visualstudio.com/DefaultCollection/DevDiv/_git/JEG-windup'
+              shell: bash
 
-  windup-build:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout WindUp upstream repo
-        uses: actions/checkout@v3
-        with:
-          repository: windup/windup
-          ref: ${{ github.base_ref }}
-      - name: Set up MS Build of OpenJDK 11
-        uses: actions/setup-java@v3
-        with:
-          java-version: '11'
-          distribution: 'microsoft'
-      - name: Cache local Maven repository
-        uses: actions/cache@v3
-        with:
-            path: ~/.m2/repository
-            key: ${{ runner.os }}-maven-azure-windup-build-${{ github.sha }}
-            restore-keys:
-                ${{ runner.os }}-maven-azure-windup-build
-      - name: Build WindUp
-        run: ./mvnw -B clean install -DskipTests
-
-  windup-rules-test:
-    runs-on: ubuntu-latest
-    needs: [windup-build]
-    strategy:
-      fail-fast: false
-      matrix:
-        tests-pattern: [ 'rules-reviewed/azure' ]
-    steps:
-      - uses: actions/checkout@v3
-      - name: Set up MS Build of OpenJDK 11
-        uses: actions/setup-java@v3
-        with:
-          java-version: '11'
-          distribution: 'microsoft'
-      - name: Cache local Maven repository
-        uses: actions/cache@v3
-        with:
-            path: ~/.m2/repository
-            key: ${{ runner.os }}-maven-azure-windup-rulesets-build-${{ github.sha }}
-            restore-keys: |
-                ${{ runner.os }}-maven-azure-windup-build-${{ github.sha }}
-      - name: Build
-        run: ./mvnw -B clean install -DskipTests
-      - name: Test
-        run: ./mvnw -B test -f rules/ -DrunTestsMatching=${{ matrix.tests-pattern }} -Dtest=WindupRulesTest
+            - name: Set up MS Build of OpenJDK 11
+              uses: actions/setup-java@v3
+              with:
+                  java-version: '11'
+                  distribution: ${{env.JAVA_DISTRIBUTION}}
```

Comment:
Q: Why not set the distribution to `microsoft` directly?

Reply:
@joe-braley - Thanks for the review. At first, my intention was to include additional distributions. However, for PR changes, it is sufficient to use just one distribution. Rather than setting the value directly in all three steps, I decided to store it in a variable.